go/vt/mysqlctl/mysqld.go

```diff
@@ -676,6 +679,32 @@ func (mysqld *Mysqld) Init(ctx context.Context, cnf *Mycnf, initDBSQLFile string
 	return nil
 }
 
+// For debugging purposes show the last few lines of the MySQL error log.
+// Return a suggestion (string) if the file is non regular or can not be opened.
+// This helps prevent cases where the error log is symlinked to /dev/stderr etc,
+// In which case the user can manually open the file.
+func readTailOfMysqldErrorLog(fileName string) string {
+	file, err := os.Open(fileName)
+	buf := make([]byte, maxLogFileSampleSize)
```

Comment:
Move this down to where it's used. We should also avoid putting any code in between a line that sets an `err` and the line that checks it.